### DISCUSSION TITLE
Migrating Netbox from Docker on unraid with Netbox v3.1.11 to production linux install v3.6.3

### DISCUSSION THREAD

**Discussion by DrDirtyDevil** [netbox-community/netbox#13996] [Help Wanted!]
Hi, 

I have used netbox for a few years now for my small company rack / combined homelab rack. I have started out on a Netbox install in docker on a Unraid host which has Version v3.1.11. I now have a VM running Linux with Netbox installed as per production documentation. i wish to copy all the content that i have gathered in the "old" netbox to my new host with the new netbox version. 

What would be the best way to do this ? 
Do i update the "old" docker first ? 
is there an Export/Import feature ?

I have tried this myself 3 times now by making a Postgres dump and replace/import that into the new environment but that does not work. 

I am a bit hesitant to update my old "Docker" because i have read a lot of people saying the docker version brakes after updating....

I have read some similar discussions but they all seem to be succesful in exporting and importing the Postgres database.
Can anyone give me some guidance ?

**Comment by candlerb** [comment]
> I have tried this myself 3 times now by making a Postgres dump and replace/import that into the new environment but that does not work.

It [does work](https://docs.netbox.dev/en/stable/administration/replicating-netbox/), and it's the safest way of doing this migration. Therefore I suggest you describe the problem you observe (e.g. what commands you typed and what errors you saw) and someone might be able to help you pin down what you're doing wrong.

> I am a bit hesitant to update my old "Docker" because i have read a lot of people saying the docker version brakes after updating....

You definitely need a backup that you're able to restore from before doing any upgrade. That is basically the same as the SQL dump/restore operation described above.

**Comment by DrDirtyDevil** [comment]
I managed to get this working. 

i had a lot of "bumps"

but in the end I simply mistyped the backup file while restoring......
netbox-backup.sql instead of backup-netbox.sql

i had to upgrade the old version on unraid first which was only slightly not working because of the 3.2 migration step of deleting contact info from the site data. I fixed that and after this it updated fine, I made a backup and restored it on the new system. 

The only reason why this took me so long was because I was IN the Postgres shell when trying to do this commands: 
psql -c 'drop database netbox'
psql -c 'create database netbox'
psql netbox < ./backup-netbox.sql

and that caused it to not spit out any errors what so ever while importing the backup. when I switched back to the bash shell of the server I got the error that netbox-backup.sql file couldnt be found but that was a simple typo because the file was actualy called backup-netbox.sql.
